### PR TITLE
Allow changing the queue of a correction

### PR DESCRIPTION
**Pull request by nadeemnazeer**
Fixes: https://github.com/sunnah-com/corrections/issues/40
- Add dropdown/button on UI
- Add update_correction_queue function
- Add action
- Get queue names dynamically on mount

cc: @hasankhan

### CODE REVIEW

**Comment by nadeemnazeer** [comment]
@hasankhan Note: When I tried to setup I too did ran into dependency conflict for the 'ecdsa' package which above test is failing for - I had to move to update base requirements with ecdsa==0.14.1. but didn't push I thought it is happening on my local only.

**Comment by hasankhan** [comment]
Yes. Let's include the dependency change as well. jazakAllah khair

**Comment by nadeemnazeer** [comment]
Done!

File: frontend/main.py [hunk 1]
```diff
@@ -153,6 +158,9 @@ def resolve_correction(queue_name, correction_id):
     elif action == 'skip':
         return skip_correction(queue_name, correction_id, version, username)
 
+    elif action == 'update_queue':
```
Comment (hasankhan) [inline frontend/main.py:161]: let's call it 'move'

File: frontend/main.py [hunk 2]
```diff
@@ -252,6 +260,17 @@ def delete_correction(queue_name, correction_id, version):
     get_correction_table().delete_item(Key={'queue': queue_name, 'id': str(
         correction_id)}, ExpressionAttributeValues={':v1': version}, ConditionExpression='version = :v1')
 
+def update_correction_queue(queue_name, correction_id, version, updated_queue_name):
+    try:
+        #it is not possible to update an attribute that forms the Key of an item thus we need to copy/delete/put in DynamoDB.
+        correction = read_correction(queue_name, correction_id, version)
```
Comment (hasankhan) [inline frontend/main.py:266]: let's read it outside the try and return not_found like skip_correction method does

File: frontend/main.py [hunk 3]
```diff
@@ -252,6 +260,17 @@ def delete_correction(queue_name, correction_id, version):
...
+        delete_correction(queue_name, correction_id, version)
+        correction['version'] = 0
+        correction['queue'] = updated_queue_name
```
Comment (hasankhan) [inline frontend/main.py:269]: you need to fix the lastAssigned as well. Let us reduce the code duplication and use a helper method in this and skip method

File: frontend/main.py [hunk 4]
```diff
@@ -131,6 +131,11 @@ def get_hadith(urn: int):
     else:
         return NotFound()
 
+@app.route('/queue_names/', methods=['GET'])
```
Comment (hasankhan) [inline frontend/main.py:134]: we can change the path to 'queues' it just so happens that we get names right now only in the response. 
the response should be a list of objects with name property

**Review by hasankhan — CHANGES_REQUESTED** [review]
JazakAllah khair. added a few comments. Can you please also attach a screenshot and squash the commits.

File: frontend/main.py (hunk 3 above)
Comment (nadeemnazeer) [reply frontend/main.py:285]: factored out a helper method to reset_correction(correction, reset_fields)

**Comment by nadeemnazeer** [comment]
@hasankhan Refined as per review, squashed the commits.
Screenshot:
image

File: frontend/main.py [hunk 5]
```diff
@@ -131,6 +131,11 @@ def get_hadith(urn: int):
     else:
         return NotFound()
 
+@app.route('/queues/', methods=['GET'])
+@aws_auth.authentication_required
+def get_queue_names():
+    return jsonify({"queue_names":["global","secondary"]})
```
Comment (hasankhan) [inline frontend/main.py:137]: Let's return a list of objects and each object having a name property. 
Let's call the method get_queues

We're returning a resource with single property.

File: frontend/main.py [hunk 6]
```diff
@@ -153,6 +158,9 @@ def resolve_correction(queue_name, correction_id):
     elif action == 'skip':
         return skip_correction(queue_name, correction_id, version, username)
 
+    elif action == 'move':
+        return update_correction_queue(queue_name, correction_id, version, data['updated_queue_name'])
```
Comment (hasankhan) [inline frontend/main.py:162]: Let's call it move_correction

File: frontend/main.py [hunk 7]
```diff
@@ -253,18 +261,41 @@ def delete_correction(queue_name, correction_id, version):
         correction_id)}, ExpressionAttributeValues={':v1': version}, ConditionExpression='version = :v1')
 
 
+def reset_correction(correction, reset_fields):
```
Comment (hasankhan) [inline frontend/main.py:264]: Let's not make the reset field configurable. Let's always reset these fields.
Please also move the helper method to the bottom.

**Review by hasankhan — CHANGES_REQUESTED** [review]

File: frontend/main.py (hunk 7 above)
Comment (nadeemnazeer) [reply frontend/main.py]: @hasankhan do we want to reset even the 'id' while moving to different queue? That is why I explicitly took the list of fields to rest.

**Comment by hasankhan** [comment]
Yes

**Review by hasankhan — APPROVED** [review]
jazakAllah khair

**Pull request merged**